src/lib/pubkey/ecies/ecies.cpp

```diff
@@ -160,6 +164,47 @@ SymmetricKey ECIES_KA_Operation::derive_secret(const std::vector<uint8_t>& eph_p
    // ISO 18033: encryption step g / decryption step i
    return SymmetricKey(kdf->derive_key(m_params.secret_length(), derivation_input));
 }
+#endif
+
+/**
+* ECIES secret derivation according to ISO 18033-2
+*/
+SymmetricKey ECIES_KA_Operation::derive_secret(const std::vector<uint8_t>& eph_public_key_bin,
+                                               const EC_AffinePoint& other_public_key_point) const {
+   if(other_public_key_point.is_identity()) {
+      throw Invalid_Argument("ECIES: peer public key point is the identity element");
+   }
```

Comment:
Nit (not sure about the formatting):

```suggestion
   BOTAN_ARG_CHECK(!other_public_key_point.is_identity(),
                   "ECIES: peer public key point is the identity element");
```